include/posix/sys/types.h

```diff
@@ -63,6 +63,19 @@ typedef u32_t clockid_t;
 typedef unsigned long timer_t;
 typedef unsigned long useconds_t;
 
+typedef u32_t pthread_rwlockattr_t;
+
+typedef struct pthread_rwlock_obj {
+	struct k_sem sem;
+	struct k_sem blocking_sem;
+	struct k_sem count_lock;
+
+	s32_t status;
+	s32_t current_reader_count;
+	s32_t blocked_reader_count;
+	k_tid_t owner;
```

Comment:
Maybe you can use the status field to signal that a rwlock is `owned`?  It's used essentially as a boolean field from what I could gather, so a bit in the status field is sufficient.

Now, I'm not sure if the status field is needed at all. It doesn't hurt being there, but since the documentation states that it's undefined to operate on an uninitialized rwlock, one could just junk it and still be compliant.  This might become a set of system calls someday, and there are checks for kernel object type and initialization state that will come in handy for this application.

Reply:
Yes as you pointed out "status" field I am using to indicate that a rwlock is valid or not. As we are not allocating space for rw_lock, so we are tracking if a lock is destroyed or not through status field only.

Yes operation on uninitialized lock in undefined so we can get rid of this. But I thought it is better to keep it to avoid bugs.

I have taken it as work size instead of boolean because operation is faster on dword.